### DISCUSSION TITLE
fwupd opens TCP and UDP ports

### DISCUSSION THREAD

**Discussion by mohemiv** [fwupd/fwupd#7170] [General]
Arch Linux.

Before fwupd:

```
$ sudo netstat -lup
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
$ sudo netstat -ltp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
```

Setting fwupd up:
```
$ sudo pacman -S fwupd
$ sudo fwupdmgr --version
```

The result:
```
$ sudo netstat -lup
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
udp        0      0 0.0.0.0:mdns            0.0.0.0:*                           108754/avahi-daemon 
udp        0      0 0.0.0.0:58195           0.0.0.0:*                           108754/avahi-daemon 
udp6       0      0 [::]:mdns               [::]:*                              108754/avahi-daemon 
udp6       0      0 [::]:47562              [::]:*                              108754/avahi-daemon 
$ sudo netstat -ltp
Active Internet connections (only servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name    
tcp        0      0 0.0.0.0:27500           0.0.0.0:*               LISTEN      108756/passimd      
```

Fwupd is a security software that is used to update vulnerable firmware. There is no way it should open 3 ports on my machine.

**Comment by hughsie** [comment]
That's not fwupd, that's passim -- I assume arch auto-installed it as a recommended package.

**Comment by mohemiv** [comment]
As far as I understand, fwupd is statically linked with passim, so it can't be installed without it.

https://archlinux.org/packages/extra/x86_64/fwupd/
```
libpassim.so=1-64 ([passim](https://archlinux.org/packages/extra/x86_64/passim/))
```

**Reply by hughsie** [reply]
In fedora we have passim-libs with the library, and the main passim package with everything else.

**Reply by hughsie** [reply]
also, if you're using systemd you can just mask the passim service -- and fwupd will happily work without it.

**Comment by mohemiv** [comment]
https://gitlab.archlinux.org/archlinux/packaging/packages/fwupd/-/issues/2
https://bugs.archlinux.org/task/79614

Arch Linux fwupd's maintainers have decided that this is "won't fix", and everyone who uses fwupd should expose TCP/UDP ports by default.

@hughsie You are the only person who can influence something to change.

**Reply by hughsie** [reply]
I have no power in the Arch packaging world! On Fedora 40+ we install and run passim by default too, although there you can remove it as a manual action.

**Reply by mohemiv** [reply]
This is a security issue. This behavior is concerning because it exposes the system to external threats. Any software should operate without opening unnecessary ports, thereby minimizing the attack surface.

CWE-1125: Excessive Attack Surface
https://cwe.mitre.org/data/definitions/1125.html

**Reply by hughsie** [reply]
> This is a security issue.

That's a pretty bold claim. The passim port needs to be open **for passim to work**. If you don't like it either remove passim or mask the passim service in systemd.

**Reply by mohemiv** [reply]
A software to update UEFI/BIOS opens four network ports (port 27500/tcp, 5353/udp, and two random UDP ports) by default, but does not need them to function.

>In Fedora, we include passim-libs with the library, and the main passim package with everything else.

As far as I understand, this is also true for Fedora.

An ordinary user should not have to delve so deeply to disable passim service. It is the user who needs passim who should enable it.

**Reply by hughsie** [reply]
> the user who needs passim who should enable it

The millions of users who should be using passim don't even know what it's called. Please read https://blogs.gnome.org/hughsie/2023/07/28/introducing-passim/ for the why.

**Reply by mohemiv** [reply]
Millions of users who use Windows have SMB and 445/tcp, but don't even know what it's called. There are reasons why other users choose Linux, and one of those reasons is not to have SMB or in our case passim.

**Reply by hughsie** [reply]
> one of those reasons is not to have SMB or in our case passim

I'm not going to argue with you any more. You have a choice to either recompile fwupd with `-Dpassim=disabled`, convince the Arch packagers to split out libpassim.so.* as a subpackage (and thus make the passim daemon removable) or you can mask the passim service in systemd.

**Reply by mohemiv** [reply]
Just to end the discussion, does Fedora's, Ubuntu's, Debian's, etc, fwupd build also opens TCP and UDP ports by default or not?

**Reply by hughsie** [reply]
Yes they do.
